File: code_folder/basics/flow_architecture/kafka_producer.py
from kafka import KafkaProducer
import yaml
import time

def load_config():
    with open('kafka_producer_config.yaml', 'r') as file:
        config = yaml.safe_load(file)
    return config

def create_message(name, metadata):
    timestamp = int(time.time())
    message = {
        'name': name,
        'timestamp': timestamp,
        'metadata': metadata
    }
    return message

def send_message(producer, topic, message):
    producer.send(topic, value=message.encode('utf-8'))
    producer.flush()

def main():
    config = load_config()
    kafka_servers = config['kafka_servers']
    topic = config['topic']
    name = config['name']
    metadata = config['metadata']

    producer = KafkaProducer(bootstrap_servers=kafka_servers)

    message = create_message(name, metadata)
    send_message(producer, topic, message)

    producer.close()

if __name__ == '__main__':
    main()

File: code_folder/basics/flow_architecture/kafka_producer.rs
use rdkafka::config::ClientConfig;
use rdkafka::producer::{FutureProducer, FutureRecord};

#[tokio::main]
async fn main() {
    let producer: FutureProducer = ClientConfig::new()
        .set("bootstrap.servers", "localhost:9092") // replace with your Kafka broker address
        .create()
        .expect("Producer creation error");

    let message_to_send = "Your message with metadata";

    let result = producer
        .send(
            FutureRecord::to("topic-name") // replace with your topic name
                .payload(message_to_send)
                .key("message-key"), // replace with your key
            5000,
        )
        .await;

    match result {
        Ok(delivery) => println!("Message delivered to {:?}", delivery),
        Err((err, _)) => println!("Error producing message: {:?}", err),
    }
}
